### PR TITLE
Frontend_with_Framework: React 7.1 -> State Management in React

### LINKED ISSUE #7590: Frontend_with_Framework: React 7.1 -> State Management in React
### Description

Contributors are required to create Documentation and explain about the **State Management in React** which should cover the questions like : 
- What is State Management in React ?
- Where we use this concept in React ?
- Benefits of using this concept ?
- also give some theory explanation with code.
- also any add-ons from your side (in-depth of the topic).

Note :

- Plagiarism is strictly prohibited.
- **Changes should be made inside the Frontend_Web_Development_React_Angular_Vue/ dir inside Frontend_Web_Development_React_Angular_Vue branch.**
- Files need to be added into the **Week 7** folder only.

### Domain

Frontend Dev React/Angular/Vue

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by nisha855**
<hr>

## Description 📜

Created documentation for State Management in React.
Fixes #7590

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [x] Frontend Dev React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by nisha855** [comment]
@heet-kumar Please review.

File: Frontend_Web_Development_React_Angular_Vue/React/Week 7/State Management in React.md
Comment (heet-kumar) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 7/State Management in React.md:48]: make this code snippets of the jsx
just in starting of the code after the 3 back-ticks add jsx (```jsx) just like this.

File: Frontend_Web_Development_React_Angular_Vue/React/Week 7/State Management in React.md
Comment (heet-kumar) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 7/State Management in React.md:95]: same here as above mention

File: Frontend_Web_Development_React_Angular_Vue/React/Week 7/State Management in React.md
Comment (heet-kumar) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 7/State Management in React.md:135]: same here also

File: Frontend_Web_Development_React_Angular_Vue/React/Week 7/State Management in React.md
Comment (heet-kumar) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 7/State Management in React.md:161]: same here also as above mentioned

File: Frontend_Web_Development_React_Angular_Vue/React/Week 7/State Management in React.md
Comment (heet-kumar) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 7/State Management in React.md:179]: same here also as above mentioned

File: Frontend_Web_Development_React_Angular_Vue/React/Week 7/State Management in React.md
Comment (heet-kumar) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 7/State Management in React.md:199]: same here also as above mentioned

File: Frontend_Web_Development_React_Angular_Vue/React/Week 7/State Management in React.md
Comment (heet-kumar) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 7/State Management in React.md:241]: same here also as above mentioned

**Review by heet-kumar — CHANGES_REQUESTED** [review]
@nisha855 just add code snippets of the jsx
just in starting of the code after the 3 back-ticks add jsx (```jsx) just like this.

**Comment by nisha855** [comment]
@heet-kumar  Done the required changes.

**Review by heet-kumar — APPROVED** [review]
@nisha855 Great work done LGTM ! 👍 lets wait for the supervisor approval and if they suggest any changes then make that to.

**Comment by heet-kumar** [comment]
@anandhu720 once review this PR from your side.

**Review by anandhu720 — APPROVED** [review]

**Pull request merged**
